packages/protocol/contracts/L1/libs/LibProposing.sol

```diff
@@ -99,42 +99,47 @@ library LibProposing {
-                // how many blocks elapsed since the last 'sync' then multiply
-                // it by 12 (L1 block time) and the reward issued per second.
-                state.slotC.accumulatedReward += uint128(
-                    (block.number - state.slotC.lastProposedHeight)
-                        * config.proposerRewardPerL1Block
-                );
-                state.slotC.lastProposedHeight = uint128(block.number);
-            }
-            state.slotC.lastProposedHeight = uint128(block.number);
-        }
+        uint128 reward;
 
-        if (
-            config.proposerRewardPerL1Block > 0 && config.proposerRewardMax > 0
-                && state.slotC.accumulatedReward > 0
-        ) {
-            // Might happen that network is congested on L1, so the pool would
-            // grow too big for the next proposeBlock() so best to have an
-            // absolute maximum upper limit.
-            reward = uint256(
-                state.slotC.accumulatedReward / config.proposerRewardHalving
-            ).min(config.proposerRewardMax);
-            // Always do halving (by config numb) even if reward is
-            // config.proposerRewardMax, so that to shrinken the pool.
-            state.slotC.accumulatedReward -=
-                state.slotC.accumulatedReward / config.proposerRewardHalving;
-            // Reward must be minted
-            LibTaikoToken.creditToken(state, resolver, msg.sender, reward, true);
+        // Reward block proposers with Taiko tokens to encourage chain adoption
+        // and ensure liveness.
+        // Rewards are issued only if `proposerRewardPerL1Block` and
+        // `proposerRewardMax` are set to nonzero values in the configuration.
+        if (config.proposerRewardPerL1Block != 0) {
+            unchecked {
+                // Mint additional tokens into the reward pool as L1 block
+                // numbers increase, to incentivize future proposers.
+                if (
+                    state.slotC.lastProposedHeight != 0
+                        && state.slotC.lastProposedHeight < block.number
+                ) {
+                    uint256 extraRewardMinted = (
+                        block.number - state.slotC.lastProposedHeight
+                    ) * config.proposerRewardPerL1Block;
+
+                    // Reward pool is capped to `proposerRewardMax`
+                    state.slotC.accumulatedReward = uint128(
+                        (extraRewardMinted + state.slotC.accumulatedReward).min(
+                            config.proposerRewardMax
+                        )
+                    );
+                }
+
+                if (state.slotC.accumulatedReward != 0) {
+                    // The current proposer receives a fixed percentage of the
+                    // reward pool.
+                    reward = state.slotC.accumulatedReward / 100
+                        * config.proposerRewardPoolPctg;
```

Comment:
Do not think it will happen often, but safer to do multiplication first then division because `state.slotC.accumulatedReward < 100` will result in 0 token.